### ISSUE TITLE
Support for filtering torrents based on streaming availability

### ISSUE THREAD

**Issue by Krenn** [erengy/taiga#441]
I would like to have Taiga ignore anything that I can watch on Crunchyroll; is that data exposed from MAL in such a way that Taiga could have a filter option for streaming site availability?

I know that an additional wrench is thrown in the works based off some things not being available in all regions on Crunchyroll, though.

**Comment by facelesspanda** [comment]
Why would you want to ignore torrents?
My guess is you are notified for each new episode available, but don't want a notification for things in your Crunchyroll queue.

The better idea is to have per-title notification settings, in my opinion. Would be easier on the dev side I think, and might work for other use-cases than this.

**Comment by erengy** [comment]
> I would like to have Taiga ignore torrents for anything that I can watch on Crunchyroll; is that data exposed from MAL in such a way that Taiga could have a filter option for streaming site availability?

MAL doesn't expose this information, unfortunately. Their API is quite limited, so no surprises here. Kitsu's API does provide [streaming links](http://docs.kitsu.apiary.io/#reference/media/streaming-links), including links for Crunchyroll. As far as I know, Kitsu gets this data from [because.moe](http://because.moe) and matches the titles to the anime in their own database.

It'd be nice to have this data for various reasons (e.g. instead of #215), but making it work for all current and future services is something I'd rather avoid at the moment. After #129, perhaps?

> The better idea is to have per-title notification settings, in my opinion.

You can already do this by creating a new filter like:

```
Name: Discard shows on Crunchyroll
Condition: Anime ID | is not | (Unknown)
Action: Discard matched items
```

You can then go on and limit this filter to any anime you'd like Taiga to ignore. It's a manual process, but it's something.

**Comment by Krenn** [comment]
@facelesspanda yes, it was to be able to keep the "available torrents" slimmed down to just those that I do have to download as I can't get them on CR.  I can set them as per-title but that's a manual process I was hoping would be avoidable via metadata, but it seems to be unavailable.

@erengy I'll take a look in to the filter like you mention.  This all certainly isn't critical or important, it's just quality-of-life polish.  Thanks for the reply!
